bin/fuel-core/src/cli/run.rs

```diff
@@ -714,6 +714,8 @@ impl Command {
                 max_tx_update_subscriptions: tx_number_active_subscriptions,
                 subscription_ttl,
                 status_cache_ttl: status_cache_ttl.into(),
+                // TODO: Where can I get this value?
+                protocol_public_key: Default::default(),
```

Comment:
I think this is a left over. If not it's not needed check to see how I implemented it without using it : https://github.com/FuelLabs/fuel-core/pull/2848

Reply:
I added this field.

Reply:
I don't understand the point of this field. I linked my PR because I did all the linking etc with this field still like that and everything is working so it doesn't seems to be useful to be set. For me the tx stauts manager should have an option and before receiving the first message from p2p it should be None and after use the key received in p2p

Reply:
The point of the delegate key is that it is delegated by a trusted key, specifically the PoA Consensus Key used for block production.

We can't just trust any signed message, we must ensure that it is the same signer as the blocks.

I'm not sure which part of your code is related to this concern.

I've removed this in favor of a `ProtocolPublicKey` port for which we have an adapter that wraps `ConsensusConfig`.

Reply:
Ah. But the new delegate keys will all be signed by this key.

Reply:
Ah yeahhhh right it didn't connected to my brain. Ty :)